ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/BlobCache.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Hyperledger Besu contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.eth.transactions;
+
+import org.hyperledger.besu.datatypes.BlobsWithCommitments;
+import org.hyperledger.besu.datatypes.VersionedHash;
+import org.hyperledger.besu.ethereum.core.Transaction;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BlobCache {
+  private final Cache<VersionedHash, BlobsWithCommitments.BlobQuad> cache;
+  private static final Logger LOG = LoggerFactory.getLogger(BlobCache.class);
+
+  public BlobCache() {
+    this.cache =
+        Caffeine.newBuilder()
+            .maximumSize(6 * 32 * 3l)
+            .build(); // 6 blobs max per 32 slots per 3 epochs
```

Comment:
configured this way the cache will always fill at its maximum before starting to evict older blobs, while you said you want 3 epochs, so instead of `maximumSize` it should be configured with `expireAfterWrite`